### DISCUSSION TITLE
[5.x] Saving deeply-nested Matrix Data via php

### DISCUSSION THREAD

**Discussion by thisanimus** [craftcms/cms#15118] [Q&A] [answered]
I noticed [this warning](https://craftcms.com/docs/5.x/reference/field-types/matrix.html#saving-matrix-fields) in the docs, and I'm curious if you can give me any hints about how this might work in Craft 5?

Would it be similar to how a standard relation would work, ie:

On Save in Parent Entry:
1. Create _Grandchild_ entry, populate it's fields
2. Create _Child_ entry, populate it's fields, add the relation to _Grandchild_
3. Add relation between _Parent_ and _Child_

**Comment by brandonkelly** [comment, marked as answer]
The [Craft 4 way](https://craftcms.com/docs/4.x/matrix-fields.html#saving-matrix-fields) still works, actually. You just need to make sure you’re using the correct entry type handles, which may have changed during the Craft 5 upgrade.

I’ll follow up with @AugustMiller and see if there’s a bigger plan for that section.

**Reply by thisanimus** [reply]
Ok, thanks
